crates/rpc-types-eth/src/transaction/receipt.rs

```diff
@@ -68,6 +68,70 @@ pub struct TransactionReceipt<T = ReceiptEnvelope<Log>> {
     pub contract_address: Option<Address>,
 }
 
+#[cfg(feature = "serde")]
+impl<'de, T> serde::Deserialize<'de> for TransactionReceipt<T>
+where
+    T: serde::Deserialize<'de>,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(serde::Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct ReceiptDeserHelper<T = ReceiptEnvelope<Log>> {
+            #[serde(flatten)]
+            inner: T,
+            transaction_hash: TxHash,
+            #[serde(default, with = "alloy_serde::quantity::opt")]
+            transaction_index: Option<u64>,
+            #[serde(default)]
+            block_hash: Option<BlockHash>,
+            #[serde(default, with = "alloy_serde::quantity::opt")]
+            block_number: Option<u64>,
+            #[serde(with = "alloy_serde::quantity")]
+            gas_used: u64,
+            // Optional fields for gas prices
+            // 1. Use effectiveGasPrice if present
+            // 2. Fallback to use gasPrice
+            // 3. Default to 0 if neither is present
+            #[serde(default, alias = "gasPrice", with = "alloy_serde::quantity::opt")]
+            effective_gas_price: Option<u128>,
+            #[serde(
+                default,
+                skip_serializing_if = "Option::is_none",
+                with = "alloy_serde::quantity::opt"
+            )]
+            blob_gas_used: Option<u64>,
+            #[serde(
+                default,
+                skip_serializing_if = "Option::is_none",
+                with = "alloy_serde::quantity::opt"
+            )]
+            blob_gas_price: Option<u128>,
+            from: Address,
+            to: Option<Address>,
+            contract_address: Option<Address>,
+        }
+
+        let helper = ReceiptDeserHelper::deserialize(deserializer)?;
+        Ok(Self {
+            inner: helper.inner,
+            transaction_hash: helper.transaction_hash,
+            transaction_index: helper.transaction_index,
+            block_hash: helper.block_hash,
+            block_number: helper.block_number,
+            gas_used: helper.gas_used,
+            effective_gas_price: helper.effective_gas_price.unwrap_or(0),
```

Comment:
ah I see we don't have access to the trait here, this seems fine

this is only affects weird networks so I don't really care here